### PR TITLE
Map creation fixes and refactoring

### PR DESCRIPTION
**Pull request by taniabogatsch**
This PR refactors the `MAP` creation code in `map.cpp`. It removes almost all special cases and uses selection vectors and our unified vector format for more uniform code paths.

### CODE REVIEW

File: src/core_functions/scalar/map/map.cpp [hunk 1]
```diff
@@ -9,211 +9,180 @@
...
+		auto values_entry = values_entries[values_idx];
 
-	auto &map_key_vector = MapVector::GetKeys(result);
-	auto &map_value_vector = MapVector::GetValues(result);
-	auto result_data = ListVector::GetData(result);
+		// validity checks
+		if (!keys_data.validity.RowIsValid(keys_idx)) {
+			MapVector::EvalMapInvalidReason(MapInvalidReason::NULL_KEY_LIST);
+		}
+		if (!keys_data.validity.RowIsValid(keys_idx)) {
```
Comment (Tishj) [inline src/core_functions/scalar/map/map.cpp:103]: values_data, values_idx

**Review by Tishj — CHANGES_REQUESTED** [review]
Thanks, looks much cleaner this way!

I saw one issue, other than that it looks good to me :)

File: src/core_functions/scalar/map/map.cpp (hunk 1 above)
Comment (taniabogatsch) [reply src/core_functions/scalar/map/map.cpp:103]: Oh, nice catch! Definitely!

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**